src/NServiceBus.AzureStoragePersistence.AcceptanceTests/Sagas/When_saga_has_duplicate_instances.cs

```diff
@@ -28,65 +27,61 @@ await Scenario.Define<Context>(c => c.OrderId = Guid.NewGuid().ToString())
-
-                        // copy saga
-                        var newId = Guid.NewGuid();
-                        saga.PartitionKey = newId.ToString();
-                        saga.RowKey = newId.ToString();
-                        saga.ETag = null;
-                        saga.Properties["Id"].GuidValue = newId;
-                        table.Execute(TableOperation.Insert(saga));
+                        var id1 = Guid.NewGuid();
+                        var id2 = Guid.NewGuid();
 
                         c.SagasIds = new[]
                         {
-                            id,
-                            newId
+                            id1,
+                            id2
                         };
 
-                        // delete the index entry making a real duplicate
-                        var indexEntry = entries.First(dte => ReferenceEquals(dte, saga) == false);
-                        table.Execute(TableOperation.Delete(indexEntry));
+                        var e1 = GetEntity(id1, c.OrderId);
+                        var e2 = GetEntity(id2, c.OrderId);
+
+                        // insert sagas duplicates
+                        table.Execute(TableOperation.Insert(e1));
+                        table.Execute(TableOperation.Insert(e2));
 
                         return session.SendLocal(new Complete
                         {
                             OrderId = c.OrderId
-                        }).ContinueWith(t => session.SendLocal(new FinalMessage()));
+                        });
                     });
                 })
                 .Done(c => c.FailedMessages.IsEmpty == false)
                 .Repeat(r => r.For(Transports.Default))
                 .Should(c =>
                 {
-                    var failedMessage = c.FailedMessages.SelectMany(kvp => kvp.Value).Single();
-                    Assert.IsInstanceOf<DuplicatedSagaFoundException>(failedMessage.Exception);
+                    CollectionAssert.IsNotEmpty(c.FailedMessages, "Should include at least one failed message.");
 
-                    foreach (var sagasId in c.SagasIds)
+                    var failedMessages = c.FailedMessages.SelectMany(kvp => kvp.Value).ToArray();
+                    foreach (var failedMessage in failedMessages)
                     {
-                        Assert.True(failedMessage.Exception.Message.Contains(sagasId.ToString()));
+                        Assert.IsInstanceOf<DuplicatedSagaFoundException>(failedMessage.Exception);
+
+                        foreach (var sagasId in c.SagasIds)
+                        {
+                            Assert.True(failedMessage.Exception.Message.Contains(sagasId.ToString()));
+                        }
                     }
                 })
                 .Run();
         }
 
+        static TwoInstanceSaga.TwoInstanceSagaEntity GetEntity(Guid id, string orderId)
```

Comment:
Can we maybe rename this to be a bit more descriptive? `GetEntity` makes me think that there's a query happening.

Perhaps `CreateSagaEntityWithOrderId`?